[2025-06-03T15:34:42.795+0200] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-06-03T15:34:42.804+0200] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: macro_trading_dag.compute_economic_quadrants manual__2025-06-03T13:31:07.164761+00:00 [queued]>
[2025-06-03T15:34:42.808+0200] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: macro_trading_dag.compute_economic_quadrants manual__2025-06-03T13:31:07.164761+00:00 [queued]>
[2025-06-03T15:34:42.808+0200] {taskinstance.py:2303} INFO - Starting attempt 2 of 3
[2025-06-03T15:34:42.822+0200] {taskinstance.py:2327} INFO - Executing <Task(SparkSubmitOperator): compute_economic_quadrants> on 2025-06-03 13:31:07.164761+00:00
[2025-06-03T15:34:42.829+0200] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'macro_trading_dag', 'compute_economic_quadrants', 'manual__2025-06-03T13:31:07.164761+00:00', '--job-id', '116', '--raw', '--subdir', 'DAGS_FOLDER/macro_trading_dag.py', '--cfg-path', '/tmp/tmpcegn8tzv']
[2025-06-03T15:34:42.830+0200] {standard_task_runner.py:91} INFO - Job 116: Subtask compute_economic_quadrants
[2025-06-03T15:34:42.835+0200] {logging_mixin.py:188} WARNING - /home/leoja/airflow/airflow_venv/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=210019) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-06-03T15:34:42.835+0200] {standard_task_runner.py:63} INFO - Started process 210051 to run task
[2025-06-03T15:34:42.863+0200] {task_command.py:426} INFO - Running <TaskInstance: macro_trading_dag.compute_economic_quadrants manual__2025-06-03T13:31:07.164761+00:00 [running]> on host N15I711-16GR512.localdomain
[2025-06-03T15:34:42.901+0200] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='macro_trading_dag' AIRFLOW_CTX_TASK_ID='compute_economic_quadrants' AIRFLOW_CTX_EXECUTION_DATE='2025-06-03T13:31:07.164761+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-03T13:31:07.164761+00:00'
[2025-06-03T15:34:42.902+0200] {taskinstance.py:430} INFO - ::endgroup::
[2025-06-03T15:34:42.905+0200] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-06-03T15:34:42.905+0200] {spark_submit.py:351} INFO - Spark-Submit cmd: spark-submit --master yarn --name compute_economic_quadrants --queue root.default /home/airflow/spark_jobs/compute_quadrants.py /home/leoja/airflow/data/Indicators.parquet /home/leoja/airflow/data/quadrants.parquet
[2025-06-03T15:34:44.092+0200] {spark_submit.py:521} INFO - 25/06/03 15:34:44 WARN Utils: Your hostname, N15I711-16GR512 resolves to a loopback address: 127.0.1.1; using 10.255.255.254 instead (on interface lo)
[2025-06-03T15:34:44.094+0200] {spark_submit.py:521} INFO - 25/06/03 15:34:44 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
[2025-06-03T15:34:44.149+0200] {spark_submit.py:521} INFO - Exception in thread "main" org.apache.spark.SparkException: When running with master 'yarn' either HADOOP_CONF_DIR or YARN_CONF_DIR must be set in the environment.
[2025-06-03T15:34:44.150+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmitArguments.error(SparkSubmitArguments.scala:650)
[2025-06-03T15:34:44.150+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateSubmitArguments(SparkSubmitArguments.scala:281)
[2025-06-03T15:34:44.150+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateArguments(SparkSubmitArguments.scala:237)
[2025-06-03T15:34:44.150+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmitArguments.<init>(SparkSubmitArguments.scala:122)
[2025-06-03T15:34:44.150+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2$$anon$3.<init>(SparkSubmit.scala:1108)
[2025-06-03T15:34:44.150+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.parseArguments(SparkSubmit.scala:1108)
[2025-06-03T15:34:44.150+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86)
[2025-06-03T15:34:44.150+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1125)
[2025-06-03T15:34:44.151+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1134)
[2025-06-03T15:34:44.151+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2025-06-03T15:34:44.171+0200] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-06-03T15:34:44.175+0200] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/leoja/airflow/airflow_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/leoja/airflow/airflow_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/leoja/airflow/airflow_venv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/leoja/airflow/airflow_venv/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/leoja/airflow/airflow_venv/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 452, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --name compute_economic_quadrants --queue root.default /home/airflow/spark_jobs/compute_quadrants.py /home/leoja/airflow/data/Indicators.parquet /home/leoja/airflow/data/quadrants.parquet. Error code is: 1.
[2025-06-03T15:34:44.177+0200] {taskinstance.py:1205} INFO - Marking task as UP_FOR_RETRY. dag_id=macro_trading_dag, task_id=compute_economic_quadrants, execution_date=20250603T133107, start_date=20250603T133442, end_date=20250603T133444
[2025-06-03T15:34:44.191+0200] {standard_task_runner.py:110} ERROR - Failed to execute job 116 for task compute_economic_quadrants (Cannot execute: spark-submit --master yarn --name compute_economic_quadrants --queue root.default /home/airflow/spark_jobs/compute_quadrants.py /home/leoja/airflow/data/Indicators.parquet /home/leoja/airflow/data/quadrants.parquet. Error code is: 1.; 210051)
[2025-06-03T15:34:44.217+0200] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-06-03T15:34:44.226+0200] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-06-03T15:34:44.226+0200] {local_task_job_runner.py:222} INFO - ::endgroup::
